Source/ASVideoPlayerNode.mm

```diff
@@ -315,14 +321,22 @@ - (NSArray *)createDefaultControlElementArray
 
 - (void)removeControls
 {
-  for (ASDisplayNode *node in [_cachedControls objectEnumerator]) {
-    [node removeFromSupernode];
+  ASLockScope(self);
+  
+  // Grab the cached controls for removing it
+  NSMutableDictionary *cachedControls = [_cachedControls copy];
+
+  {
+    ASUnlockScope(self);
+    for (ASDisplayNode *node in [cachedControls objectEnumerator]) {
+      [node removeFromSupernode];
+    }
   }
 
-  [self cleanCachedControls];
+  [self __locked_cleanCachedControls];
```

Comment:
I think it's safe to do this before calling `removeFromSupernode` on cached controls? If yes we can avoid the unlock scope.